vm_eval.c

```diff
@@ -936,6 +936,34 @@ rb_funcallv_kw(VALUE recv, ID mid, int argc, const VALUE *argv, int kw_splat)
     return rb_call(recv, mid, argc, argv, kw_splat ? CALL_FCALL_KW : CALL_FCALL);
 }
 
+/*!
+ * Calls a method only if it is the basic method of `ancestor`
+ * otherwise returns Qundef;
+ * \param recv   receiver of the method
+ * \param mid    an ID that represents the name of the method
+ * \param mid    the Class that defined the basic method
+ * \param argc   the number of arguments
+ * \param argv   pointer to an array of method arguments
+ * \param kw_splat bool
+ */
+VALUE
+rb_funcallv_basic_kw(VALUE recv, ID mid, VALUE ancestor, int argc, const VALUE *argv, int kw_splat)
+{
+    const rb_callable_method_entry_t *cme;
+    rb_execution_context_t *ec;
+    VALUE klass = CLASS_OF(recv);
+    if(!klass) return Qundef; /* hidden object? */
+
+    cme = rb_callable_method_entry(klass, mid);
+    if(cme && METHOD_ENTRY_BASIC(cme) && RBASIC_CLASS(cme->defined_class) == ancestor) {
```

Comment:
Would you put a space?

```suggestion
    if (cme && METHOD_ENTRY_BASIC(cme) && RBASIC_CLASS(cme->defined_class) == ancestor) {
```